### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by chunfantasy**
This PR is due to adding documentation for this project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+### Application mode
+
+The dongle is in application mode if it has an application to run and is simply plugged in,
+or after a successful DFU operation.
+
+In application mode the visible USB device depends on the application firmware.
+For further documentation please refer to the [Nordic SDK]().
+
+In application mode it is **expected** that the visible USB device has a _DFU trigger interface_.
+This interface provides a `semver` string which identifies the application firmware currently running,
```
Comment (bihanssen) [inline README.md]: Complex sentence, suggest rewriting, example: 
This interface provides a `semver` string which identifies the application firmware currently running.
If the `semver` doesn't match the expected value, the device will be reset into bootloader mode.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+### Build
+
+The project is using `rollup.js`, so the following command is needed to run the build:
+    npm run rollup
+
+
+### Test
+
+The project comes with automated integration tests in the `test` directory. In order to run the test, nRF52840 development kit must be attached to the PC on CDC ACM port or nRF52840 dongle must be attached to the PC.
+Meanwhile, the environment variable `DFU_SERIAL_NUMBER` need to be set to specify a certain nRF52840 dongle.
```
Comment (bihanssen) [inline README.md]: Suggest rewording: "Meanwhile" can be removed. /need/needs/.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+Or if it is not specified, then the first found nRF52840 dongle by `SerialOPort` will be used for the test.
```
Comment (bihanssen) [inline README.md]: /Serial0Port/SerialPort/

**Comment by chunfantasy** [comment]
Nice comments and suggestions 👍
